plugins/repoquery.py

```diff
@@ -260,22 +282,72 @@ def run(self, args):
 
         # filter repo and arch
         q = self.filter_repo_arch(opts, q)
+        orquery = q
 
         if opts.file:
             q = q.filter(file=opts.file)
         if opts.whatprovides:
             q = self.by_provides(self.base.sack, [opts.whatprovides], q)
-        if opts.whatrequires:
+        if opts.alldeps:
+            if not opts.whatrequires:
+                print("\n--alldeps requires --whatrequires option.")
+                print("Example: dnf repoquery --whatrequires audiofile --alldeps")
+                return
```

Comment:
And one more thing. We need to raise an exception here to return a non-zero exit status.
So, can you please replace the `print`s and the `return` with e.g.

```
raise dnf.exceptions.Error(
    _("--alldeps requires --whatrequires option."
      "Example: dnf repoquery --whatrequires audiofile --alldeps"))
```

?

FYI, the ```_()``` function translates the string to the user's preferred language.